### PR TITLE
chore: use W3C webdriver for saucelabs browser tests

### PR DESCRIPTION
**Pull request by devcorpio**
# Summary

Before the changes we were launching the Sauce Lab tests using [JWP](https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol). With these changes we will start using [W3C WebDriver Protocol](https://docs.saucelabs.com/dev/glossary/#w3c-webdriver-protocol).

In the image above we can see how the tests are being executed through the W3C protocol

Screenshot 2021-12-14 at 16 11 05

Details to have into account:

- Chrome versions have been updated to newer ones.
- The rest of browsers versions have been configured to meet the requirements of W3C protocols, more info [here](https://docs.saucelabs.com/dev/w3c-webdriver-capabilities/#browser-compatibility).

# Next steps

- ~~In the investigation/exploration done in this task we have tried to tackle the update of all the @wdio dependencies but it deserves its own issue since several errors arose in the process.~~ (Finally included in this PR)

### CODE REVIEW

File: dev-utils/karma.js [hunk 1]
```diff
@@ -89,9 +89,9 @@ const baseConfig = {
     startConnect: false,
     recordVideo: false,
     recordScreenshots: true,
-    tunnelIdentifier,
+    tunnelName,
     options: {
-      // seleniumVersion: '2.48.2',
+      seleniumVersion: '2.48.2',
```
Comment (devcorpio) [inline dev-utils/karma.js:94]: @vigneshshanmugam do you know why this line was commented?

Comment (vigneshshanmugam) [reply dev-utils/karma.js:94]: With older selenium versions, it wasnt possible to launch latest browsers. We can remove this versioning deps.

File: package.json [hunk 2]
```diff
@@ -178,6 +178,7 @@
     "vue": "^3.2.20",
     "vue-loader": "^16.8.1",
     "vue-router": "^4.0.12",
+    "vue-template-compiler": "^2.6.10",
```
Comment (vigneshshanmugam) [inline package.json:181]: Dont think this is relevant, we have moved to vue 3 where this shouldnt be required.

**Review by vigneshshanmugam — COMMENTED** [review]
Other changes looks good to me.

File: package-lock.json [hunk 3]
```diff
@@ -9068,45 +9068,98 @@
         "yarn-install": "^1.0.0"
       },
       "dependencies": {
+        "@wdio/config": {
```
Comment (vigneshshanmugam) [inline package-lock.json:9071]: not sure how the lockfiles got changed? Did you try to upgrade wdio deps?

File: package.json (hunk 2 above)
Comment (devcorpio) [reply package.json]: I removed it, it was a leftover 😓

File: package-lock.json (hunk 3 above)
Comment (devcorpio) [reply package-lock.json:9071]: yes, a leftover of a frustrated attempt to upgrade wdio deps 😄  thanks for catching it

File: dev-utils/integration-test.js [hunk 4]
```diff
@@ -44,6 +44,11 @@ async function runIntegrationTest(pageUrl) {
     })
 
     await page.goto(pageUrl, { timeout: 30000 })
+    await page.evaluate(() => {
+      // From puppeteer 2.0 onwards the Chromium included with it supports the CompressionStream api.
+      // Because of that, the postData property is set to undefined causing the test to fail
+      CompressionStream = void 0
+    })
```
Comment (devcorpio) [inline dev-utils/integration-test.js:51]: @vigneshshanmugam, this one has been tricky b to find:

**Puppeteer request having Compression API enabled:**

`{ hasPostData: true, postData: undefined }`

**Puppeteer request having Compression API disabled:**

`{ hasPostData: true, postData: [...data...] }`

File: packages/rum/test/e2e/async-tests/async.e2e-spec.js [hunk 5]
```diff
@@ -50,13 +50,19 @@ describe('async-tests', function () {
...
+      'external',
+      'longtask',
+      'app',
+      'resource'
+    ]
     /**
-     * Safari does not support Resource & User Timing API
+     * Safari version configured in the test report span type browser-timing
      */
     const { name } = getBrowserInfo()
```
Comment (vigneshshanmugam) [inline packages/rum/test/e2e/async-tests/async.e2e-spec.js:63]: Seems like you would need to change this function to return the correct browser name and version with the new updated capabilities?

File: dev-utils/integration-test.js (hunk 4 above)
Comment (vigneshshanmugam) [reply dev-utils/integration-test.js:51]: Good catch, Yes that should explain the integration test failure on rum package.

File: packages/rum/test/e2e/async-tests/async.e2e-spec.js (hunk 5 above)
Comment (devcorpio) [reply packages/rum/test/e2e/async-tests/async.e2e-spec.js:63]: The update to make use of the new capabilities is within that function:

`That function returns the values properly:

`
function getBrowserInfo() {
  const { browserVersion, browserName } = browser.capabilities
  return {
    name: browserName.toLowerCase(),
    version: browserVersion
  }
}
``

File: packages/rum/test/e2e/async-tests/async.e2e-spec.js [hunk 6]
```diff
@@ -50,13 +50,19 @@ describe('async-tests', function () {
...
-    if (name.indexOf('safari') === -1) {
-      spanTypes.push('resource', 'app')
+    if (name.indexOf('safari') !== -1) {
+      spanTypes.push('browser-timing')
```
Comment (vigneshshanmugam) [inline packages/rum/test/e2e/async-tests/async.e2e-spec.js:65]: This is interesting, span types should be hard-navigation.browser-timing. Not sure why we are seeing this behavior. 

https://github.com/elastic/apm-agent-rum-js/blob/0862d55a36cc0f906b61ff3289d7fa7416ce4d0a/packages/rum-core/src/performance-monitoring/capture-navigation.js#L98

**Review by vigneshshanmugam — APPROVED** [review]
Thanks @devcorpio 🎉 , I know this has been a painful debugging experience, But glad we did this now.

File: packages/rum/test/e2e/async-tests/async.e2e-spec.js (hunk 6 above)
Comment (devcorpio) [reply packages/rum/test/e2e/async-tests/async.e2e-spec.js]: You are right, probably it was a bad interpretation from my side. The test works properly with all the new updates, I removed the safari logic completely 🥳

**Pull request merged**